hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -245,14 +246,17 @@ private synchronized FileSystemViewManager getViewManager() {
   public abstract HoodieWriteMetadata<O> insertOverwriteTable(HoodieEngineContext context, String instantTime, I records);
 
   /**
-   * update statistics info for current table.
-   * to do adaptation, once RFC-27 is finished.
+   * Updates Metadata Indexes (like Z-Index)
+   * TODO rebase onto metadata table (post RFC-27)
    *
-   * @param context HoodieEngineContext
-   * @param instantTime Instant time for the replace action
-   * @param isOptimizeOperation whether current operation is OPTIMIZE type
-   */
-  public abstract void updateStatistics(HoodieEngineContext context, List<HoodieWriteStat> stats, String instantTime, Boolean isOptimizeOperation);
+   * @param context instance of {@link HoodieEngineContext}
+   * @param instantTime instant of the carried operation triggering the update
+   */
+  public abstract void updateMetadataIndexes(
```

Comment:
rename: updateColumnStats? for now, this is not connected to the metadata table at all and even then would only update one partition.  So kind of misleading

Reply:
This method actually calls into `updateColumnStats` in #4106. Will inline it there